### PR TITLE
feat: [PL-24645]: Hotfix - azure repo connector (#8751)

### PR DESCRIPTION
**Pull request by mayankVermaHarness**
##### Summary:

Hotfix for https://github.com/harness/harness-core-ui/pull/8751/

This form will use StepWizard and have multiple Steps such as:

ConnectorDetailsStep
GitDetailsStep
StepAzureRepoAuthentication
DelegateSelectorStep
VerifyOutOfClusterDelegateStep

##### Jira Links:

https://harness.atlassian.net/browse/PL-24645

##### Screenshots:

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

**Review by abhinavrastogi-harness — DISMISSED** [review]

**Review by avinashmadhwani02 — APPROVED** [review]

File: src/modules/35-connectors/components/CreateConnector/AzureRepoConnector/CreateAzureRepoConnector.module.scss [hunk 1]
```diff
@@ -0,0 +1,12 @@
+/*
+ * Copyright 2021 Harness Inc. All rights reserved.
+ * Use of this source code is governed by the PolyForm Shield 1.0.0 license
+ * that can be found in the licenses directory at the root of this repository, also available at
+ * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
+ */
+
+.azureRepoConnector {
```
Comment (sunnykesh-harness) [inline src/modules/35-connectors/components/CreateConnector/AzureRepoConnector/CreateAzureRepoConnector.module.scss:8]: Icon colour should be controlled by props.

File: src/modules/35-connectors/components/CreateConnector/AzureRepoConnector/StepAuth/StepAzureRepoAuthentication.module.scss [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+      }
+    }
+
+    .authHeaderRow {
+      display: flex;
+      justify-content: space-between;
+      align-items: baseline;
+      :global {
+        .bp3-input-group {
+          .bp3-input {
```
Comment (sunnykesh-harness) [inline src/modules/35-connectors/components/CreateConnector/AzureRepoConnector/StepAuth/StepAzureRepoAuthentication.module.scss:30]: Now we have done standardisation exercise, if possible avoid custom stylings of these form input elements.

File: src/modules/35-connectors/components/CreateConnector/AzureRepoConnector/StepAuth/StepAzureRepoAuthentication.tsx [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+  const authOptions: Array<SelectOption> = [
+    {
+      label: getString('usernameToken'),
+      value: GitAuthTypes.USER_TOKEN
+    }
+  ]
+
+  useEffect(() => {
+    if (loadingConnectorSecrets && props.isEditMode) {
+      /* istanbul ignore else */
```
Comment (sunnykesh-harness) [inline src/modules/35-connectors/components/CreateConnector/AzureRepoConnector/StepAuth/StepAzureRepoAuthentication.tsx:148]: Why we need /`* istanbul ignore else */` here?

File: src/modules/35-connectors/pages/connectors/views/savedDetailsView/SavedConnectorDetails.tsx [hunk 4]
```diff
@@ -737,7 +737,8 @@ const getSchemaByType = (
     case Connectors.GITHUB:
     case Connectors.GITLAB:
     case Connectors.BITBUCKET:
-      return getGithubSchema(connector) // GitHub schema will work for GitLab, Bitbucket too
+    case Connectors.AZURE_REPO:
+      return getGithubSchema(connector) // GitHub schema will work for GitLab, Bitbucket and AzureRepos too
```
Comment (sunnykesh-harness) [inline src/modules/35-connectors/pages/connectors/views/savedDetailsView/SavedConnectorDetails.tsx:741]: Good to update comment for AZURE_REPO

**Review by sunnykesh-harness — APPROVED** [review]

**Review by abhinavrastogi-harness — APPROVED** [review]

**Pull request merged**